src/lib/Controller/api/abstract_camera.c

```diff
@@ -138,13 +138,15 @@ void abstract_camera_toggle_remote(WbDevice *d, WbRequest *r) {
 }
 
 bool abstract_camera_request_image(AbstractCamera *ac, const char *functionName) {
-  double current_simulation_time = wb_robot_get_time();
+  const double current_simulation_time = wb_robot_get_time();
+  const double previous_image_update_time = ac->image_update_time;  // in case of reset, time can go backward
   if (ac->image_update_time >= current_simulation_time)
     return true;
 
   ac->image_requested = true;
   wb_robot_flush_unlocked();
-  if (ac->image_update_time != current_simulation_time && robot_is_quitting() == 0) {
+  if (ac->image_update_time != current_simulation_time && previous_image_update_time <= ac->image_update_time &&
```

Comment:
The new condition seems to be always true, isn't it?

Reply:
No, because the `wb_robot_flush_unlocked` function will increase the `ac->image_update_time` time, therefore `ac->image_update_time == current_simulation_time`, and about `previous_image_update_time <= ac->image_update_time`, this is not true in case of reset, because `ac->image_update_time == 0.0`.

Reply:
OK, I see. Thanks for the explanation.